integration-tests/src/test_loop/utils/trie_sanity.rs

```diff
@@ -374,14 +397,37 @@ pub fn check_state_shard_uid_mapping_after_resharding(
             assert!(value.is_none());
             continue;
         }
-        let parent_value = store.get(parent_shard_uid, &node_hash);
-        // Parent shard data must still be accessible using parent ShardUId.
+        let parent_value = trie_store.get(parent_shard_uid, &node_hash);
+        // Sanity check: parent shard data must still be accessible using Trie interface and parent ShardUId.
         assert_eq!(&parent_value.unwrap()[..], value.unwrap());
+
         // All parent shard data is available via both children shards.
-        for child_shard_uid in &children_shard_uids {
-            let child_value = store.get(*child_shard_uid, &node_hash);
+        for child_shard_uid in &tracked_mapped_children {
+            let child_value = trie_store.get(*child_shard_uid, &node_hash);
             assert_eq!(&child_value.unwrap()[..], value.unwrap());
         }
+        checked_any_key = true;
+    }
+    assert!(checked_any_key);
+
+    let shards_tracked_after_resharding =
+        get_tracked_shards_from_prev_block(client, resharding_block_hash);
+    // Sanity checks if the node tracks all shards (e.g. it is RPC node).
+    if !client.config.tracked_shards.is_empty() {
+        assert_eq!(tracked_mapped_children.len(), 2);
+        assert_eq!(
+            shards_tracked_after_resharding.len(),
+            epoch_config.shard_layout.num_shards() as usize
+        );
+    }
+    // If any child shard was tracked after resharding, it means the node had to split the parent shard.
```

Comment:
Wondering what happens if the node tracks the parent both not any child

We do resharding for nothing? 🤔

Reply:
Yes, added a test in fa8e29e23c81